src/keeper.rs

```diff
@@ -0,0 +1,104 @@
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, RwLock,
+    },
+    thread::{Builder, JoinHandle},
+};
+
+use solana_client::tpu_client::TpuClient;
+use solana_quic_client::{QuicConfig, QuicConnectionManager, QuicPool};
+use solana_sdk::{
+    compute_budget::ComputeBudgetInstruction, hash::Hash, instruction::Instruction,
+    message::Message, signature::Keypair, signer::Signer, transaction::Transaction,
+};
+
+use crate::{helpers::to_sdk_instruction, states::PerpMarketCache};
+
+fn create_root_bank_update_instructions(perp_markets: &Vec<PerpMarketCache>) -> Vec<Instruction> {
+    perp_markets
+        .iter()
+        .map(|perp_market| {
+            let ix = mango::instruction::update_root_bank(
+                &perp_market.mango_program_pk,
+                &perp_market.mango_group_pk,
+                &perp_market.mango_cache_pk,
+                &perp_market.root_bank,
+                perp_market.node_banks.as_slice(),
+            )
+            .unwrap();
+            to_sdk_instruction(ix)
+        })
+        .collect()
+}
+
+fn create_update_price_cache_instructions(perp_markets: &Vec<PerpMarketCache>) -> Vec<Instruction> {
+    perp_markets
+        .iter()
+        .map(|perp_market| {
+            let ix = mango::instruction::cache_prices(
+                &perp_market.mango_program_pk,
+                &perp_market.mango_group_pk,
+                &perp_market.mango_cache_pk,
+                perp_market.node_banks.as_slice(),
+            )
+            .unwrap();
+            to_sdk_instruction(ix)
+        })
+        .collect()
+}
+
+pub fn send_transaction(
+    tpu_client: Arc<TpuClient<QuicPool, QuicConnectionManager, QuicConfig>>,
+    ixs: &[Instruction],
+    blockhash: Arc<RwLock<Hash>>,
+    payer: &Keypair,
+) {
+    let mut tx = Transaction::new_unsigned(Message::new(ixs, Some(&payer.pubkey())));
+    if let Ok(recent_blockhash) = blockhash.read() {
+        tx.sign(&[payer], *recent_blockhash);
+    }
```

Comment:
It is somewhat suspicious that the locking error is ignored.
As this is a CLI application, it probably does not need to be able to handle abnormal termination of threads.
In which case, it is commonly written as just:

```suggestion
    let recent_blockhash = blockhash.read().unwrap();
    tx.sign(&[payer], *recent_blockhash);
```